dashboard/app/models/level.rb

```diff
@@ -179,9 +179,13 @@ def self.level_cache
   end
 
   def update_level_in_cache
-    return unless Script.should_cache?
-    Level.level_cache[id] = self if Level.level_cache.key? id
-    Level.level_cache[name] = self if Level.level_cache.key? name
+    return unless Script.should_cache? && Level.level_cache.key?(id)
```

Comment:
Don't we want to refresh the cache on edit? Doesn't the latter conjunction prevent updates to the cache?